StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Views/PromoBadgeView.swift

```diff
@@ -89,7 +134,15 @@ class PromoBadgeView: UIView {
         ])
     }
     
-    private func formatPromoText(_ text: String) -> String {
+    private func formatPromoText(_ text: String, eligible: Bool) -> String {
+        guard eligible else {
+            let baseString = STPLocalizedString(
+                "No %@ promo",
+                "Label for when the user is not eligible for a promo."
+            )
```

Comment:
Should we do the same thing here where we only show the full text if the locale is set to english? Otherwise it'd be great to have a snapshot test of this in a different locale. I'm mostly curious of what the horizontal overflow state looks like

Reply:
I added a snapshot test for a long bank name, which results in an overflow. I’ll bring up the concerns around text that’s too long with the team.